src/client/java/teammates/client/scripts/DataMigrationForSanitizedDataInStudentProfileAttributes.java

```diff
@@ -96,6 +97,6 @@ private void desanitizeAndUpdateProfile(StudentProfileAttributes profile)
     private boolean hasAnySanitizedField(StudentProfileAttributes profile) {
         return isSanitizedHtml(profile.shortName) || isSanitizedHtml(profile.email)
                 || isSanitizedHtml(profile.institute) || isSanitizedHtml(profile.nationality)
-                || isSanitizedHtml(profile.gender) || isSanitizedHtml(profile.moreInfo);
+                || isSanitizedHtml(String.valueOf(profile.gender)) || isSanitizedHtml(profile.moreInfo);
```

Comment:
You can remove isSanitizedHtml(String.valueOf(profile.gender)) because it'll always return false so it uselless in that case.